lib/model.js

```diff
@@ -2284,35 +2291,38 @@ class Model {
 
         return [instance, true];
       }).catch(sequelizeErrors.UniqueConstraintError, err => {
-        const flattenedWhere = Utils.flattenObjectDeep(options.where);
-        const flattenedWhereKeys = Object.keys(flattenedWhere).map(name => _.last(name.split('.')));
-        const whereFields = flattenedWhereKeys.map(name => _.get(this.rawAttributes, `${name}.field`, name));
-        const defaultFields = options.defaults && Object.keys(options.defaults)
-          .filter(name => this.rawAttributes[name])
-          .map(name => this.rawAttributes[name].field || name);
-
-        if (defaultFields) {
-          if (!_.intersection(Object.keys(err.fields), whereFields).length && _.intersection(Object.keys(err.fields), defaultFields).length) {
-            throw err;
-          }
-        }
 
-        if (_.intersection(Object.keys(err.fields), whereFields).length) {
-          _.each(err.fields, (value, key) => {
-            const name = this.fieldRawAttributesMap[key].fieldName;
-            if (value.toString() !== options.where[name].toString()) {
-              throw new Error(`${this.name}#findOrCreate: value used for ${name} was not equal for both the find and the create calls, '${options.where[name]}' vs '${value}'`);
+        if (err.fields) { 
+          const flattenedWhere = Utils.flattenObjectDeep(options.where);
+          const flattenedWhereKeys = Object.keys(flattenedWhere).map(name => _.last(name.split('.')));
```

Comment:
I think you could instead write a quick util to find the last segment after <char> in a string rather than creating an array and throwing it away right away.

Reply:
It was there before. I took out `flattenedWhereKeys` out of the equation, it was redundant.